### PR TITLE
EAR-551 Sort out destination picker

### PR DESCRIPTION
**Pull request by tjamesmac**
## Useful links
[Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-551)
[Designs](https://overflow.io/s/96LRCTYQ?node=2d584288&prototype)

### What is the context of this PR?
Sorting out the destination content picker.

It should:
- Allow users to pick questions from the same sections
- Provide only the sections if they are routing outside of the current section
- Provide other destinations if any of the above don't work

### How to review
- Create questionnaire
- Add four question pages in section one
- Add an additional 3 sections
- Go to S1 q2 and create a routing rule
- Click `select a destination` and the modal should display
NOTE: Design of modal being fixed in another ticket
- Should be three tabs on the left (`Current`, `Later`, `Other`)
- Each tab should display correct destination values
- If clicking the `Else` destination selector, `Later` should not appear
NOTE: should be able to switch tabs and choose a destination with `Space` or `Enter`
- Confirm the selection and check it appears in the destination picker

### CODE REVIEW

File: eq-author/src/App/page/Logic/Routing/DestinationSelector/RoutingDestinationContentPicker.js [hunk 1]
```diff
@@ -144,48 +80,50 @@ export const UnwrappedRoutingDestinationContentPicker = ({
...
+  });
+
+  return (
+    <>
+      <UnwrappedRoutingDestinationContentPicker
+        data={data}
+        loading={loading}
+        {...otherProps}
+      />
+    </>
```
Comment (tbaggaley) [inline eq-author/src/App/page/Logic/Routing/DestinationSelector/RoutingDestinationContentPicker.js:124]: Is the fragment needed?

Also - do we still need to have two components here? We could just have RoutingDestinationContentPicker and do away with the wrapped/unwrapped distinction now you're using useQuery - or was it to make testing easier?

```suggestion
  return (
      <UnwrappedRoutingDestinationContentPicker
        data={data}
        loading={loading}
        {...otherProps}
      />
```

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 2]
```diff
@@ -56,35 +56,38 @@ export const ParentMenuItem = styled(MenuItem)`
...
-      opacity: 1;
-    }
-    &:hover {
-      background: #f3f3f3;
-    }
-  }
+  ${props =>
+    props["aria-selected"] &&
+    css`
+      background-color: #f3f3f3;
```
Comment (tbaggaley) [inline eq-author/src/components/ContentPickerv2/Menu.js:62]: Is there a colour constant we can use while you're changing this bit?

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 3]
```diff
@@ -56,35 +56,38 @@ export const ParentMenuItem = styled(MenuItem)`
...
+      &::after {
+        opacity: 1;
+      }
+      &:hover {
+        background: #f3f3f3;
```
Comment (tbaggaley) [inline eq-author/src/components/ContentPickerv2/Menu.js:67]: Le same

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 4]
```diff
@@ -56,35 +56,38 @@ export const ParentMenuItem = styled(MenuItem)`
...
+    props["aria-selected"] &&
+    css`
+      background-color: ${colors.primary};
+      --color: ${colors.white};
+      --colorSecondary: ${colors.white};
 
-    background: ${colors.primary};
-    border-color: #377090;
-    z-index: 2;
+      border-color: #377090;
```
Comment (tbaggaley) [inline eq-author/src/components/ContentPickerv2/Menu.js:84]: Le same

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 5]
```diff
@@ -56,35 +56,38 @@ export const ParentMenuItem = styled(MenuItem)`
...
+      z-index: 2;
 
-    &:hover {
-      background: #397596;
-    }
-  }
+      &:hover {
+        background: #397596;
```
Comment (tbaggaley) [inline eq-author/src/components/ContentPickerv2/Menu.js:88]: Le same

**Review by tbaggaley — CHANGES_REQUESTED** [review]
Hey Tom, some super warm and fuzzy refactoring going on, excellent!

Just a couple small comments from code review. Haven't tried in Author yet

File: eq-author/src/components/ContentPickerv2/Menu.js (hunk 4 above)
Comment (jasehumphr) [reply eq-author/src/components/ContentPickerv2/Menu.js:86]: Let's not hard code them innit

Comment (jasehumphr) [reply eq-author/src/components/ContentPickerv2/Menu.js:86]: We currently don't have that in the theme but it's a blue so I guess. ${colors.blue} would do it

**Review by farres1 — DISMISSED** [review]

File: eq-author/src/App/page/Logic/Routing/DestinationSelector/RoutingDestinationContentPicker.js (hunk 1 above)
Comment (tjamesmac) [reply eq-author/src/App/page/Logic/Routing/DestinationSelector/RoutingDestinationContentPicker.js]: Removed unwrapped too

**Review by jasehumphr — APPROVED** [review]
S.W.E.E.T.

**Review by tbaggaley — APPROVED** [review]
Superfantastic

**Review by farres1 — APPROVED** [review]

**Pull request merged**